### DISCUSSION TITLE
during `publisher confirm`,will the message in queue get pushed to consumer even before the `publisher confirm` notification is sent?

### DISCUSSION THREAD

**Discussion by Divine1** [rabbitmq/rabbitmq-server#2664] [Other] [answered]
i was looking into [this documentation](https://www.rabbitmq.com/confirms.html) and i came across below scenario mentioned in the documentation. 

![image](https://user-images.githubusercontent.com/10482177/100862443-2b00d100-34b9-11eb-844b-5486e1f1eeb3.png)

Please let me know if my below understanding is correct.

1. A client(publisher) sends a persistent message to a durable queue.
2. A client(consumer) consumes the message from the queue but confirms are not active and due to some reason the rabbitmqserver gets restarted
3. The client(consumer) reconnects to the queue in rabbitmq server and starts consuming messages again

**my query is**
will the clientConsumer receive the message even before the `publisherConfirm` notification is sent to the clientPublisher?
if yes, why doesn't rabbitmq  handle the scenario in this way?  **incase of a `persistent message` and a `durable queue`, the clientConsumer will receive the messages only after `publisher confirm` was sent for that particular message (this means that the message is persisted into the disk ?**

The scenario mentioned in [this documentation](https://www.rabbitmq.com/confirms.html) doesn't make any sense to me especially when it can be handled by the software itself. 

kindly clarify

**Comment by michaelklishin** [comment, marked as answer]
Your understanding of the documentation seems to be a bit off. The section quoted talks about *publisher confirm delivery on the publishing channel*. Consumers are not involved in any way, in fact, there may be no consumers online.

Delivery to consumers is not coordinated with publisher confirm delivery back to the publishers in any way. Confirms will typically be sent first but that's not guaranteed.

**Reply by Divine1** [reply]
@michaelklishin  i understand now. thank you for the response :)
